### PR TITLE
[14.0][MIG] rma_account

### PR DESCRIPTION
**Pull request by MateuGForgeFlow**
Standard migration to v14.
Depends on #146.

cc @AaronHForgeFlow

### CODE REVIEW

**Comment by max3903** [comment]
@MateuGForgeFlow Any reason for deleting the rma module in this last commit?

**Comment by JordiBForgeFlow** [comment]
@MateuGForgeFlow it must be a mistake. Please revert this commit back.

**Comment by HviorForgeFlow** [comment]
Do not panic! :smile:

**Comment by JordiBForgeFlow** [comment]
image

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Thanks for migrating this!! Good job! Overall looks good. But there are a couple of bugs in some places:

* The originating invoice line field is not filtering by Product, there must be something incorrect in the domain:

image

* When clicking in "Originating Invoice" button, and then clicking on the invoice, there is an error related to an invalid domain.
image

* When creating the refund, I got a Validation Error because of the accounts of the product. I did some debugging and it came out that the product was not in the wizard line, thus, no product, no accounts, and fails. There must something incorrect in the field declaration
image

Regards.

File: rma_account/views/account_move_view.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <field name="invoice_vendor_bill_id" position="after">
+                 <field
+                    name="add_rma_line_id"
+                    context="{'rma': True}"
+                    domain="[('partner_id', '=', partner_id)]"
+                    attrs="{'readonly': [('state','not in',['draft'])],
+                         'invisible': ['|', ('state', '=', 'paid')]}"
```
Comment (max3903) [inline rma_account/views/account_move_view.xml]: @MateuGForgeFlow You are missing an argument or you should remove the operator...

**Comment by max3903** [comment]
@MateuGForgeFlow The domain on the "Originating Invoice Line" is showing up journal entries from the payment:
Screenshot from 2021-01-14 10-41-36

**Comment by MateuGForgeFlow** [comment]
@max3903 Sorry, this is still WIP, I will let you know when it is ready for review.
Thank you.

**Comment by max3903** [comment]
@MateuGForgeFlow Fine with me but can you provide an ETA? Can you do next Wednesday?

**Comment by MateuGForgeFlow** [comment]
@max3903 Perfect, you'll have it by then.

**Comment by HviorForgeFlow** [comment]
@MiquelRForgeFlow I think we've already face this problem. Do you remember the solution?, I guess was adding a domain to only fetch move line of invoice type, but I don't remember where we can find an example.

Thanks!

File: rma_account/views/rma_order_line_view.xml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                >
+                        <field name="refund_count" widget="statinfo" string="Refunds" />
+                    </button>
+            </button>
+            <group name="main_info" position="inside">
+                <field
+                    name="account_move_line_id"
+                    options="{'no_create': True}"
+                    context="{'rma': True}"
+                    domain="['|',
```
Comment (MiquelRForgeFlow) [inline rma_account/views/rma_order_line_view.xml]:
```suggestion
                    domain="[('move_id.type', 'not in', ('entry','out_invoice','out_refund')), '|',
```

File: rma_account/views/rma_order_line_view.xml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
```
Comment (MiquelRForgeFlow) [inline rma_account/views/rma_order_line_view.xml]:
```suggestion
                    domain="[('move_id.type', '!=', 'entry'), '|',
```

**Review by MiquelRForgeFlow — COMMENTED** [review]
@HviorForgeFlow something like this?

File: rma_account/views/rma_order_line_view.xml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                                ('move_id.partner_id', '=', partner_id),
+                                ('move_id.partner_id', 'child_of', partner_id)]"
```
Comment (HviorForgeFlow) [inline rma_account/views/rma_order_line_view.xml:34]:
```suggestion
                    domain="['&', ('move_id.type', 'not in', ('entry','out_invoice','out_refund')), '|',
                                ('move_id.partner_id', '=', partner_id),
                                ('move_id.partner_id', 'child_of', partner_id)]"
```

File: rma_account/views/rma_order_line_view.xml (hunk 2 above)
Comment (HviorForgeFlow) [reply rma_account/views/rma_order_line_view.xml]: yes but an array, not a tuple after not in

**Review by MateuGForgeFlow — COMMENTED** [review]
@HviorForgeFlow In Odoo 14 "type" changes to "move_type"

**Comment by HviorForgeFlow** [comment]
> @HviorForgeFlow In Odoo 14 "type" changes to "move_type"

Good catch :rocket:

**Review by AaronHForgeFlow — APPROVED** [review]
For my side it is fine to merge as first version. @max3903  I think it is ok to review.

File: rma_account/views/account_move_view.xml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                        </page>
+                    </notebook>
+                </field>
+            </data>
+        </field>
+    </record>
+
+
+    <record id="view_invoice_customer_rma_form" model="ir.ui.view">
+        <field name="name">account.move.customer.rma</field>
```
Comment (max3903) [inline rma_account/views/account_move_view.xml]:
```suggestion
    <record id="view_invoice_supplier_rma_form" model="ir.ui.view">
        <field name="name">account.move.supplier.rma</field>
```

**Comment by MateuGForgeFlow** [comment]
@max3903 Comments attended. Anything else to solve before merge? 
Thank you.

**Comment by JordiBForgeFlow** [comment]
@MateuGForgeFlow can you rebase the other PR's?

**Pull request merged**